.github/workflows/reusable-on-code-change.yml

```diff
@@ -0,0 +1,198 @@
+name: Reusable Workflow (On Code Change)
+on:
+  workflow_call:
+    inputs:
+        static_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
```

Comment:
This will most likely soon be an array (see: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on). Could you check how this would be used?

I understand we're using something similar (no concern yet for array) on our other reusable workflows.

Reply:
The array syntax is only for self hosted runners and when we have several types of runners and we need to precise a type of runner to use. If I remember well we only have one type of self hosted runner so we shouldn't be concerned by this

Reply:
Sorry, I meant we will likely soon have multiple types of runners.

How will we handle that in such a case?